jbmc/unit/java-testing-utils/require_goto_statements.cpp

```diff
@@ -295,6 +294,54 @@ const code_declt &require_goto_statements::require_declaration_of_name(
   throw no_decl_found_exceptiont(variable_name.c_str());
 }
 
+static bool is_allocate(const exprt &expr)
+{
+  auto side_effect_expr = expr_try_dynamic_cast<side_effect_exprt>(expr);
+  return side_effect_expr && side_effect_expr->get_statement() == ID_allocate;
+}
+
+static const irep_idt &
+get_name_of_symbol_with_struct_component_ptr_type(
```

Comment:
I'm not sure either the name or comments match what this does? It appears to expect an assignment a = b, then go looking for either b = ALLOCATE(...) or b = c, where c is again a symbol? Either way I can't see how struct_component_ptr_types are involved?

Reply:
Your summary is correct, except it's `a->x = b`. The fact that the returned symbol (`b` or `c`) has the correct type comes after the call to `get_name_of_symbol_with_struct_component_ptr_type`. So it's more like `get_name_of_symbol_which_we_believe_will_have_struct_component_ptr_type`. The function doesn't check that it has that type in any way.